plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -1091,6 +1093,27 @@ public String quoted(RemoteTableName remoteTableName)
         return emptyMap();
     }
 
+    @Override
+    public OptionalLong delete(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        QueryBuilder queryBuilder = new QueryBuilder(this);
+        JdbcTableHandle tableHandle = (JdbcTableHandle) handle;
+        checkArgument(tableHandle.isNamedRelation(), "Unable to delete from synthetic table: %s", tableHandle);
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            connection.setAutoCommit(false);
+            PreparedQuery preparedQuery = queryBuilder.prepareDelete(session, connection, tableHandle.getRequiredNamedRelation(), tableHandle.getConstraint());
+            OptionalLong update;
+            try (PreparedStatement preparedStatement = queryBuilder.prepareStatement(session, connection, preparedQuery)) {
+                update = OptionalLong.of(preparedStatement.executeUpdate());
+            }
+            connection.commit();
+            return update;
```

Comment:
will it return `null` if try fails?

Reply:
No. You won't get here as exception would be thrown.